pkg/reconciler/v1alpha1/revision/resources/deploy.go

```diff
@@ -158,6 +150,28 @@ func makePodSpec(rev *v1alpha1.Revision, loggingConfig *logging.Config, observab
 	return podSpec
 }
 
+func getUserPort(rev *v1alpha1.Revision) int32 {
+	if len(rev.Spec.Container.Ports) == 1 {
+		return rev.Spec.Container.Ports[0].ContainerPort
+	}
+
+	return v1alpha1.RevisionContainerUserPortDefaultValue
+}
+
+func buildContainerPorts(userPort int32) []corev1.ContainerPort {
```

Comment:
We would want to take in both the port name as well.

Reply:
See above.

Reply:
Yup, my mistake -- please ignore.